app/javascript/containers/PortfolioListContainer.jsx

```diff
@@ -1,37 +1,84 @@
-import React from 'react';
+import React from "react";
+import ReactModal from "react-modal";
 
-import { loadInitialState } from '../actions/initialState';
-import { bindActionCreators } from 'redux';
-import { connect } from 'react-redux';
-import { Link } from 'react-router-dom';
+import { loadInitialState } from "../actions/initialState";
+import { bindActionCreators } from "redux";
+import { connect } from "react-redux";
+import { Link } from "react-router-dom";
+import { post } from "../fetch/requester";
+import { addBuildingType } from "../actions/building_type";
 
 class PortfolioListContainer extends React.Component {
-  componentDidMount() {
-    if (window.INITIAL_STATE) {
-      this.props.initActions.loadInitialState(window.INITIAL_STATE);
+  constructor(props) {
+    super(props);
+    this.state = {
+      showModal: false
+    };
+    this.toggleModal = this.toggleModal.bind(this);
+    this.createBuildingType = this.createBuildingType.bind(this);
+  }
+
+  toggleModal() {
+    this.setState({ showModal: !this.state.showModal });
+  }
+
+  async createBuildingType(event) {
+    event.preventDefault();
+    const typeName = event.target.name.value;
+    try {
+      let response = await post("/api/building_types", {
+        name: typeName,
+        categories: []
+      });
+      const buildingType = { ...response.data, questions: [] };
+      const buildingTypeId = buildingType.id;
+      this.props.addBuildingType(buildingType);
+      this.props.history.push(`/building_types/${buildingTypeId}`);
+    } catch (error) {
+      console.log(error);
```

Comment:
Instead of logging the error to the console, can we show the error text in the modal (maybe above the submit button)?